document/js-api/JS.bs

```diff
@@ -0,0 +1,922 @@
+</div>
+
+<div algorithm>
+  The <dfn method for="WebAssembly">instantiate(bytes, importObject)</dfn> method, when invoked, performs the following steps:
+    1. Let |stableBytes| be a [=get a copy of the buffer source|copy of the bytes held by the buffer=] |bytes|.
+    1. Let |promiseOfModule| be [=a new promise=].
+    1. [=Asynchronously compile a WebAssembly module=] from |stableBytes| with |promiseOfModule|.
+    1. [=Instantiate a promise of a module|Instantiate=] |promiseOfModule| with imports |importObject| and return the result.
+</div>
+
+<div algorithm>
+  The <dfn method for="WebAssembly">instantiate(moduleObject, importObject)</dfn> method, when invoked, performs the following steps:
+    1. Let |promise| be [=a new promise=].
+    1. [=Queue a task=] to perform the following steps:
+        1. [=instantiate a WebAssembly module|Instantiate the WebAssembly module=] |module| importing |importObject|, and let |instance| be the result.  If this throws an exception, catch it, and [=reject=] |promise| with the exception.
+        1. [=Resolve=] |promise| with |instance|.
+    1. Return |promise|
+</div>
+
+Note: A follow-on streaming API is documented [in the WebAssembly design repository](https://github.com/WebAssembly/design/blob/master/Web.md#additional-web-embedding-api).
+
+<h3 id="modules">Modules</h3>
+
+<pre class="idl">
+enum ImportExportKind {
+  "function",
+  "table",
+  "memory",
+  "global"
+};
+
+dictionary ModuleExportDescriptor {
+  required DOMString name;
+  required ImportExportKind kind;
+  // Note: Other fields such as signature may be added in the future.
+};
+
+dictionary ModuleImportDescriptor {
+  required DOMString module;
+  required DOMString name;
+  required ImportExportKind kind;
+};
+
+[LegacyNamespace=WebAssembly, Constructor(BufferSource bytes), Exposed=(Window,Worker,Worklet)]
+interface Module {
+  static sequence&lt;ModuleExportDescriptor> exports(Module module);
+  static sequence&lt;ModuleImportDescriptor> imports(Module module);
+  static sequence&lt;ArrayBuffer> customSections(Module module, DOMString sectionName);
+};
+</pre>
+
+<div algorithm>
+  The <dfn>string value of the extern type</dfn> |type| is
+    * "function" if |type| is of the form 𝖿𝗎𝗇𝖼 functype
+    * "table" if |type| is of the form 𝗍𝖺𝖻𝗅𝖾 tabletype
+    * "memory" if |type| is of the form 𝗆𝖾𝗆    memtype
+    * "global" if |type| is of the form 𝗀𝗅𝗈𝖻𝖺𝗅 globaltype
+</div>
+
+<div algorithm>
+    The <dfn method for="Module">exports(moduleObject)</dfn> method, when invoked, performs the following steps:
+    1. Let |module| be |moduleObject|.\[[Module]].
+    1. Let |exports| be an empty [=list=].
+    1. For each (|nameBytes|, |type|) in [=module_exports=](|module|)
+        1. Let |name| be |nameBytes| [=UTF-8 decode without BOM or fail|decoded as UTF-8=].
+        1. Assert: The previous operation did not fail, as |module| is [=valid=].
```

Comment:
This should be something like: "Assert: _name_ is not failure (_module_ is valid)."

Reply:
Same below.

Reply:
Done